packages/parser/src/schemas/cognito.ts

```diff
@@ -0,0 +1,522 @@
+  }),
+  response: z.object({}),
+});
+
+/**
+ * A zod schema for a Cognito Pre-Authentication trigger event.
+ *
+ * @example
+ * ```json
+ * {
+ *   "request": {
+ *     "userAttributes": {
+ *       "email": "user@example.com",
+ *       "name": "John Doe"
+ *     },
+ *     "validationData": {
+ *       "someKey": "someValue"
+ *     },
+ *     "userNotFound": false
+ *   },
+ *   "response": {}
+ * }
+ * ```
+ *  * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-authentication.html}
+ */
+export const PreAuthenticationTriggerSchema = z.object({
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    validationData: z.record(z.string(), z.string()),
+    userNotFound: z.boolean(),
+  }),
+  response: z.object({}),
+});
+
+/**
+ * A zod schema for a Cognito Post-Authentication trigger event.
+ *
+ * @example
+ * ```json
+ * {
+ *   "request": {
+ *     "userAttributes": {
+ *       "email": "user@example.com",
+ *       "name": "John Doe"
+ *     },
+ *     "newDeviceUsed": true,
+ *     "clientMetadata": {
+ *       "customKey": "customValue"
+ *     }
+ *   },
+ *   "response": {}
+ * }
+ * ```
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-post-authentication.html}
+ */
+export const PostAuthenticationTriggerSchema = z.object({
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    newDeviceUsed: z.boolean(),
+    clientMetadata: z.record(z.string(), z.string()).optional(),
+  }),
+  response: z.object({}),
+});
+
+/**
+ * A zod schema for a Cognito Pre-Token Generation trigger event (version 1).
+ *
+ * @example
+ * ```json
+ * {
+ *   "request": {
+ *     "userAttributes": { "string": "string" },
+ *     "groupConfiguration": {
+ *       "groupsToOverride": [ "string", "string" ],
+ *       "iamRolesToOverride": [ "string", "string" ],
+ *       "preferredRole": "string"
+ *     },
+ *     "clientMetadata": { "string": "string" }
+ *   },
+ *   "response": {
+ *     "claimsOverrideDetails": {
+ *       "claimsToAddOrOverride": { "string": "string" },
+ *       "claimsToSuppress": [ "string", "string" ],
+ *       "groupOverrideDetails": {
+ *         "groupsToOverride": [ "string", "string" ],
+ *         "iamRolesToOverride": [ "string", "string" ],
+ *         "preferredRole": "string"
+ *       }
+ *     }
+ *   }
+ * }
+ * ```
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html}
+ */
+export const PreTokenGenerationTriggerSchemaV1 = z.object({
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    groupConfiguration: z.object({
+      groupsToOverride: z.array(z.string()),
+      iamRolesToOverride: z.array(z.string()),
+      preferredRole: z.string(),
+    }),
+    clientMetadata: z.record(z.string(), z.string()),
+  }),
+  response: z.object({
+    claimsOverrideDetails: z.object({
```

Comment:
This entire field should just be `z.nullable()` since the field must be populated as part of the request.